src/main/java/edu/kit/datamanager/repo/util/validators/impl/HandleNetValidator.java

```diff
@@ -0,0 +1,156 @@
+        log.debug("Prefix: {}", prefix);
+        log.debug("Suffix: {}", suffix);
+        ignoringExc(() -> {
+            try{
+                fullValid.set(urlValidator.isValid(serverAddress + "/" + prefix + "/" + suffix));
+            } catch (Exception ignored){
+            }
+        });
+        if (fullValid.get()) {
+            LOG.info("The handle is valid!");
+            return true;
+        }
+        log.warn("Either the suffix or the prefix might be invalid. Proving if prefix is valid...");
+        if (urlValidator.isValid("https://hdl.handle.net/0.NA/" + prefix)) {
+            log.info("The prefix {} is valid!", prefix);
+            throw new MessageValidationException("Prefix valid, but suffix not");
+        }
+        log.error("The entered prefix is invalid!");
+        throw new BadArgumentException("Prefix not provable on handle.net");
+    }
+
+    /**
+     * Uses the other isDownloadable method with the handle.net server as serverAddress
+     *
+     * @param prefix the handle prefix
+     * @param suffix the handle suffix
+     * @return true if the record is downloadable
+     */
+    private boolean isDownloadable(String prefix, String suffix) {
+        return isDownloadable("http://hdl.handle.net/api/handles", prefix, suffix);
+    }
+
+    /**
+     * This method uses regex to get the prefix, suffix and the server address out of a http or https URL.
+     * After this it uses the isDownloadable method to check whether the input is downloadable.
+     *
+     * @param url to validate
+     * @return true if the input is valid and downloadable.
+     */
+    private boolean isValidHTTPURL(String url) {
+        String regex = "(http|https)://(.+)/([A-Za-z0-9.]+)/([A-Za-z0-9.]+)";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(url);
+        if (matcher.find()) {
+            return isDownloadable((matcher.group(1) + "://" + matcher.group(2)), matcher.group(3), matcher.group(4));
+        } else throw new BadArgumentException("Invalid input");
+    }
+
+    /**
+     * This method uses regex to check whether unallowed characters are used and to extract the prefix and suffix.
+     * After this it uses the isDownloadable method to check whether the input is downloadable.
+     *
+     * @param handle to validate
+     * @return true if the input is valid and downloadable.
+     */
+    private boolean isValidHandle(String handle) {
+        String regex = "([A-Za-z0-9.]+)/([A-Za-z0-9.]+)";
+        Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        Matcher matcher = pattern.matcher(handle);
+        if (matcher.find()) {
+            return isDownloadable(matcher.group(1), matcher.group(2));
+        } else throw new BadArgumentException("Invalid input");
+    }
+
+    private static void ignoringExc(Runnable r) {
+        try {
+            r.run();
+        }
+        catch (Exception ignored) {
+        }
```

Comment:
I don't like empty catch blocks. At least a message at debug level should be logged.

Reply:
In this case the exception is unimportant and shouldn't be processed by anything.